lightning_pose/utils/scripts.py

```diff
@@ -135,14 +137,37 @@ def get_data_module(
 ) -> Union[BaseDataModule, UnlabeledDataModule]:
     """Create a data module that splits a dataset into train/val/test iterators."""
 
+    # Old configs may have num_gpus: 0. We will remove support in a future release.
+    if cfg.training.num_gpus == 0:
+        warnings.warn(
+            "Config contains unsupported value num_gpus: 0. "
+            "Update num_gpus to 1 in your config."
+        )
+    cfg.training.num_gpus = max(cfg.training.num_gpus, 1)
+
     semi_supervised = check_if_semi_supervised(cfg.model.losses_to_use)
     if not semi_supervised:
-        if not (cfg.training.gpu_id, int):
-            raise NotImplementedError("Cannot fit fully supervised model on multiple gpus")
+        # Divide config batch_size by num_gpus to maintain the same effective batch
+        # size in a multi-gpu setting.
+        if cfg.training.train_batch_size % cfg.training.num_gpus != 0:
+            raise ValidationError(
+                "train_batch_size should be a multiple of num_gpus. "
+                "train_batch_size=%d, num_gpus=%d"
+                % (cfg.training.train_batch_size, cfg.training.num_gpus)
+            )
+        if cfg.training.val_batch_size % cfg.training.num_gpus != 0:
+            raise ValidationError(
+                "val_batch size should be a multiple of num_gpus. "
```

Comment:
small stylistic thing, I'd like to eventually update the codebase to only use f-strings, they're cleaner in my opinion, e.g.
```
f"val_batch_size={cfg.training.val_batch_size}, num_gpus={cfg.training.num_gpus}"
```

Reply:
Ah, didn't know about that. Cool feature. Updated to use that.